test/TestFilePathEvaluation.py

```diff
@@ -0,0 +1,149 @@
+"""Testing file path evaluation when using import_tasks / include_tasks."""
+import textwrap
+
+import pytest
+
+from ansiblelint.runner import Runner
+
+LAYOUT_IMPORTS = {
+    'directories': ['tasks', 'tasks/subtasks'],
+    'files': {
+        'main.yml': textwrap.dedent(
+            """\
+            ---
+            - hosts: target
+              gather_facts: false
+              tasks:
+                - name: from main import task 1
+                  import_tasks: tasks/task_1.yml
+            """
+        ),
+        'tasks/task_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 1 import task 2
+              import_tasks: tasks/task_2.yml
+            """
+        ),
+        'tasks/task_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 2 import subtask 1
+              import_tasks: tasks/subtasks/subtask_1.yml
+            """
+        ),
+        'tasks/subtasks/subtask_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 1 import subtask 2
+              import_tasks: tasks/subtasks/subtask_2.yml
+            """
+        ),
+        'tasks/subtasks/subtask_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 2 do something
+              debug:
+                msg: |
+                  Something...
+            """
+        ),
+    },
+}
+
+LAYOUT_INCLUDES = {
+    'directories': ['tasks', 'tasks/subtasks'],
+    'files': {
+        'main.yml': textwrap.dedent(
+            """\
+            ---
+            - hosts: target
+              gather_facts: false
+              tasks:
+                - name: from main import task 1
+                  include_tasks: tasks/task_1.yml
+            """
+        ),
+        'tasks/task_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 1 import task 2
+              include_tasks: tasks/task_2.yml
+            """
+        ),
+        'tasks/task_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from task 2 import subtask 1
+              include_tasks: tasks/subtasks/subtask_1.yml
+            """
+        ),
+        'tasks/subtasks/subtask_1.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 1 import subtask 2
+              include_tasks: tasks/subtasks/subtask_2.yml
+            """
+        ),
+        'tasks/subtasks/subtask_2.yml': textwrap.dedent(
+            """\
+            ---
+            - name: from subtask 2 do something
+              debug:
+                msg: |
+                  Something...
+            """
+        ),
+    },
+}
+
+
+@pytest.fixture
+def create_root(tmp_path):
+    """Define root path for project."""
+    return str(tmp_path)
```

Comment:
There's no need to convert everything into strings everywhere. It's wise versa — everything should be kept as pathlib objects and only converted in a specific place where this is needed (although, it's probably unnecessary at all).